### PR TITLE
[Modules] Deployment Script module: adding support for using existing storage account

### PR DESCRIPTION
**Pull request by lsnoddy**
# Description

Added a storageAccountId parameter and storageAccountSettings  to add support for the use of an existing storage account:

https://learn.microsoft.com/en-us/azure/templates/Microsoft.Resources/2020-10-01/deploymentScripts?pivots=deployment-language-bicep#azurepowershellscriptproperties

| Pipeline |
| - |
| |
[Resources - DeploymentScripts](https://github.com/lsnoddy/ResourceModules/actions/workflows/ms.resources.deploymentscripts.yml)

# Type of Change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 1]
```diff
@@ -105,6 +113,7 @@ resource deploymentScript 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
     azPowerShellVersion: kind == 'AzurePowerShell' ? azPowerShellVersion : null
     azCliVersion: kind == 'AzureCLI' ? azCliVersion : null
     containerSettings: empty(containerGroupName) ? null : containerSettings
+    storageAccountSettings: empty(storageAccountId) ? null : storageAccountSettings
     arguments: arguments
     environmentVariables: empty(environmentVariables) ? null : environmentVariables
     scriptContent: empty(scriptContent) ? null : scriptContent
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:119]:
```suggestion
    containerSettings: !empty(containerGroupName) ? containerSettings : null
    storageAccountSettings: !empty(storageAccountId) ? storageAccountSettings : null
    arguments: arguments
    environmentVariables: !empty(environmentVariables) ? environmentVariables : null
    scriptContent: !empty(scriptContent) ? scriptContent : null
```

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 2]
```diff
@@ -52,6 +52,9 @@ param cleanupPreference string = 'Always'
 @description('Optional. Container group name, if not specified then the name will get auto-generated. Not specifying a \'containerGroupName\' indicates the system to generate a unique name which might end up flagging an Azure Policy as non-compliant. Use \'containerGroupName\' when you have an Azure Policy that expects a specific naming convention or when you want to fully control the name. \'containerGroupName\' property must be between 1 and 63 characters long, must contain only lowercase letters, numbers, and dashes and it cannot start or end with a dash and consecutive dashes are not allowed.')
 param containerGroupName string = ''
 
+@description('Optional. The storage account ID.')
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:55]:
```suggestion
@description('Optional. The resource ID of the storage account to use for this deployment script. If none is provided, the deployment script uses a temporary, managed storage account.')
```

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 3]
```diff
@@ -52,6 +52,9 @@ param cleanupPreference string = 'Always'
...
+param storageAccountId string = ''
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:56]:
```suggestion
param storageAccountResourceId string = ''
```

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 4]
```diff
@@ -83,6 +86,11 @@ var identity = identityType != 'None' ? {
   userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
 } : null
 
+var storageAccountSettings = {
+  storageAccountKey: !empty(storageAccountId) ? listKeys(storageAccountId, '2019-06-01').keys[0].value : null
+  storageAccountName: !empty(storageAccountId) ? last(split(storageAccountId, '/')) : null
+}
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:92]:
```suggestion
var storageAccountSettings = !empty(storageAccountResourceId) ? {
  storageAccountKey: listKeys(storageAccountResourceId, '2019-06-01').keys[0].value
  storageAccountName: last(split(storageAccountResourceId, '/'))
} : {}
```

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 5]
```diff
@@ -105,6 +113,7 @@ resource deploymentScript 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
     azPowerShellVersion: kind == 'AzurePowerShell' ? azPowerShellVersion : null
     azCliVersion: kind == 'AzureCLI' ? azCliVersion : null
     containerSettings: empty(containerGroupName) ? null : containerSettings
+    storageAccountSettings: empty(storageAccountId) ? null : storageAccountSettings
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:116]:
```suggestion
    storageAccountSettings: empty(storageAccountResourceId) ? null : storageAccountSettings
```

File: settings.yml
Comment (AlexanderSehr) [inline settings.yml:1]: Beware, this file has to be rolled back

File: modules/Microsoft.Resources/deploymentScripts/readme.md
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/readme.md:1]: Must be updated if the other suggestions are applied

File: settings.yml
Comment (lsnoddy) [reply settings.yml:1]: Rolled back

File: modules/Microsoft.Resources/deploymentScripts/readme.md
Comment (lsnoddy) [reply modules/Microsoft.Resources/deploymentScripts/readme.md:1]: Updated readme to reflect latest changes

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep (hunk 3 above)
[the remark this answers is not in the document]
Comment (lsnoddy) [reply modules/Microsoft.Resources/deploymentScripts/deploy.bicep]: Agreed, added storage account to cli and ps test cases

File: modules/Microsoft.Resources/deploymentScripts/deploy.bicep [hunk 6]
```diff
@@ -104,10 +112,11 @@ resource deploymentScript 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
...
-    containerSettings: empty(containerGroupName) ? null : containerSettings
+    containerSettings: !empty(containerGroupName) ? containerSettings : null
+    storageAccountSettings: !empty(storageAccountResourceId) ? storageAccountSettings : null
     arguments: arguments
-    environmentVariables: empty(environmentVariables) ? null : environmentVariables
-    scriptContent: empty(scriptContent) ? null : scriptContent
+    environmentVariables: !empty(environmentVariables) ? environmentVariables : null
+    scriptContent: !empty(scriptContent) ? scriptContent : null
     primaryScriptUri: empty(primaryScriptUri) ? null : primaryScriptUri
     supportingScriptUris: empty(supportingScriptUris) ? null : supportingScriptUris
```
Comment (AlexanderSehr) [inline modules/Microsoft.Resources/deploymentScripts/deploy.bicep:121]:
```suggestion
    primaryScriptUri: !empty(primaryScriptUri) ? primaryScriptUri: null
    supportingScriptUris: !empty(supportingScriptUris) ? supportingScriptUris : null
```

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**